src/python/pants/core/util_rules/external_tool.py

```diff
@@ -175,6 +224,35 @@ def get_request(self, plat: Platform) -> ExternalToolRequest:
             f"{self.known_versions}"
         )
 
+    @classmethod
+    def check_version_constraints(
+        cls, version: str, constraints: Requirement, action: UnsupportedVersionUsage
+    ) -> None:
+        if constraints.specifier.contains(version):  # type: ignore[attr-defined]
+            # all ok
+            return None
+
+        msg = [
+            f"The option [{cls.options_scope}].version is set to {version}, which is not compatible with what this release of Pants expects: {constraints}",
+        ]
+
+        if action is UnsupportedVersionUsage.Allow:
+            msg.append(
+                f"The option [{cls.options_scope}].use_unsupported_version is set to 'allow'."
+            )
+            logger.info("\n".join(msg))
+        else:
+            msg.append(
+                "Please update the version to a supported value, or consider using a different Pants release if you cannot change the version."
```

Comment:
Formatting nit: Black does not format strings because it doesn't want to risk breaking them. We try to stick to 100 columns or less, and this is a lot more. You would need to manually fix this by using a new line. We use implicit string concacatenation a lot here:

```python
x = (
    "a sentence. "
    "another sentence."
)
```

Note the space at the end of the first line.

--

It looks like a couple strings in this PR are over 100 columns.

Reply:
yeah, guess I thought black would help fix those too... I'll go over them.